lib/std/os/uefi/posix.zig

```diff
@@ -0,0 +1,743 @@
+    uts.release[release.len] = 0;
+
+    @memcpy(&uts.version, "2.0.0");
+    uts.version[5] = 0;
+
+    const machine = @tagName(builtin.cpu.arch);
+    @memcpy(uts.machine[0..machine.len], machine);
+    uts.machine[machine.len] = 0;
+}
+
+pub fn unlinkat(dirfd: fd_t, file_path: []const u8, flags: u32) std.posix.UnlinkatError!void {
+    switch (dirfd) {
+        .file => |p| {
+            var path_buffer: [PATH_MAX_WIDE]u16 = undefined;
+            const len = try std.unicode.wtf8ToWtf16Le(&path_buffer, file_path);
+            path_buffer[len] = 0;
+
+            const fd = p.open(path_buffer[0..len :0], .{ .write = true }, .{}) catch |err| switch (err) {
+                error.NotFound => return error.FileNotFound,
+                error.NoMedia => return error.FileSystem,
+                error.MediaChanged => return error.FileSystem,
+                error.DeviceError => return error.FileSystem,
+                error.VolumeCorrupted => return error.FileSystem,
+                error.WriteProtected => return error.AccessDenied,
+                error.AccessDenied => return error.AccessDenied,
+                error.OutOfResources => return error.SystemResources,
+                else => |e| return unexpectedError(e),
+            };
+            errdefer fd.close();
+
+            const stat = try fstat(.{ .file = fd });
+
+            // fd is a directory and AT_REMOVEDIR is not set
+            if (stat.mode & S.IFDIR != 0 and flags & AT.REMOVEDIR != 0)
+                return error.IsDir;
+
+            if (!fd.delete()) {
+                // delete failed, likely because this is a directory and not empty
+                return error.DirNotEmpty;
+            }
+        },
+        .simple_output => return error.NotDir,
+        .simple_input => return error.NotDir,
+        .none => return error.NotDir,
+        .cwd => return unlinkat(uefi.working_directory, file_path, flags),
+    }
+}
+
+pub fn write(fd: fd_t, buf: []const u8) std.posix.WriteError!usize {
+    switch (fd) {
+        .file => |p| {
+            var size: usize = buf.len;
+            p.write(&size, buf.ptr).err() catch |err| switch (err) {
+                error.Unsupported => return error.NotOpenForWriting,
+                error.NoMedia => return error.InputOutput,
+                error.DeviceError => return error.InputOutput,
+                error.VolumeCorrupted => return error.InputOutput,
+                error.WriteProtected => return error.NotOpenForWriting,
+                error.AccessDenied => return error.AccessDenied,
+                else => |e| return unexpectedError(e),
+            };
+            return size;
+        },
+        .simple_output => |p| {
+            const view = std.unicode.Utf8View.init(buf) catch unreachable;
+            var iter = view.iterator();
+
+            // rudimentary utf16 writer
+            var index: usize = 0;
+            var utf16: [256]u16 = undefined;
+            while (iter.nextCodepoint()) |rune| {
```

Comment:
Probably better to avoid the term `rune`.

Reply:
Renamed to `codepoint`

Reply:
`codepoint` is a good name, it is coming from `nextCodepoint` after all.